src/react-components/input/Slider.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import { useState, useCallback, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+import styles from "./Slider.scss";
+import { trackColor } from "./Slider.scss";
+
+function round(step, n) {
+  return Math.round(n / step) * step;
+}
+
+function getLinearGradientCSS(ratio, leftColor, rightColor) {
+  return [
+    "-webkit-gradient(",
+    "linear, ",
+    "left top, ",
+    "right top, ",
+    "color-stop(" + ratio + ", " + leftColor + "), ",
+    "color-stop(" + ratio + ", " + rightColor + ")",
+    ")"
+  ].join("");
+}
+
+function updateGradient(el, min, max, value) {
+  if (el) {
+    const ratio = ((value - min) / (max - min)).toFixed(2);
+    el.style.backgroundImage = getLinearGradientCSS(ratio, trackColor, "transparent");
+  }
+}
+
+export function Slider({ min, max, step, digits, value, defaultValue, onChange, disabled, className }) {
+  const myRoot = useRef();
+  const inputRef = useRef();
+  const [displayValue, setDisplayValue] = useState(-1);
+  const [isDragging, setIsDragging] = useState(false);
+
+  let currentValue = defaultValue;
+  if (isDragging) {
+    currentValue = displayValue !== -1 ? displayValue : defaultValue;
+  } else {
+    currentValue = value !== undefined ? value : displayValue !== -1 ? displayValue : defaultValue;
+  }
+
+  const stopDragging = useCallback(
+    () => {
+      setIsDragging(false);
+    },
+    [setIsDragging]
+  );
+
+  const onUpdateValue = useCallback(
+    v => {
+      setDisplayValue(v);
+      if (onChange) {
+        onChange(v);
+      }
+    },
+    [onChange, setDisplayValue]
+  );
+
+  const drag = useCallback(
```

Comment:
Confused why we have to manually implement all this drag logic.. Doesn't the range input handle this for us? Something to do with styling?

Reply:
I think I just copy pasted this from the prefs code and didn't really questioned it that much. Looking at it now I think we can get rid of a bunch of code. I'll test it in different platforms, it might have been there for a reason...